demo-opensuse/README.md

```diff
@@ -0,0 +1,248 @@
+
+For the sake of demonstrating in-toto, we will have you run all parts of the software supply chain. That is, you will perform the commands on behalf of Alice, Bob and Carl as well as the client who verifies the final product.
+
+# Setup
+Install docker
+https://docs.docker.com/engine/installation/
+
+Get in-toto
+```
+git clone https://github.com/in-toto/in-toto.git
+```
+
+Build Docker image from Dockerfile
+```
+cd in-toto/demo-opensuse/
+docker pull opensuse
+docker build -t="in-toto-demo-opensuse" .
+```
+Run the docker image
+```
+docker run --privileged -i -t in-toto-demo-opensuse
+```
+
+# Change into the demo directoy and you are ready to start
+```
+cd home/in-toto/
+source in-toto/bin/activate/
+cd demo-opensuse/
+```
+
+Inside the demo-opensuse directory you will find four directories: `owner_alice`,
+`functionary_bob`, `functionary_carl` and `final_product`. Alice, Bob and Carl
+already have RSA keys in each of their directories. This is what you see:
+```
+tree  # If you don't have tree, try 'find .' instead
+# the tree command gives you the following output
+# ├── README.md
+# ├── Dockerfile
+# ├── final_product
+# ├── functionary_bob
+# │   ├── bob
+# │   └── bob.pub
+# ├── functionary_carl
+# │   ├── carl
+# │   └── carl.pub
+# └── owner_alice
+#     ├── alice
+#     ├── alice.pub
+#     └── create_layout.py
+```
+
+### Define software supply chain layout (Alice)
+First, we will create a new package on open build server and we will need to define the software supply chain layout. To simplify this process, we provide a script that generates a simple layout for the purpose of the demo.
+
+In this software supply chain layout, we have Alice, who is the project owner that creates the layout, Bob, who clones the project's repo and performs some pre-packaging editing (update changelog), Carl, then tests the build and verifies that its fit to ship, Carl then commits the package which triggers open build server to build the RPMs.
+
+First, signup for an account on opensuse build service at
+https://build.opensuse.org/ICSLogin/
+
+Export your username as environment variable so that you can copy paste the demo commands
+```
+username=<username>
+```
+
+Checkout your default project. You will be prompted for your username and password which is stored in `~/.oscrc` as plain text.
+```
+cd owner_alice/
+osc checkout home:$username
+```
+
+Specify the build target
+```
+osc meta prj -e home:$username
+```
+
+This will open a template xml in your favourite editor. For this demo uncomment the first one.
+```
+<repository name="openSUSE_Factory">
+   <path project="openSUSE:Factory" repository="standard" />
+   <arch>x86_64</arch>
+   <arch>i586</arch>
+ </repository>
```

Comment:
I'm thinking that these operations should also be performed by a functionary (maybe bob?)